bundles/org.openhab.binding.openweathermap/src/main/java/org/openhab/binding/openweathermap/internal/connection/OpenWeatherMapConnection.java

```diff
@@ -239,6 +243,61 @@ public OpenWeatherMapConnection(OpenWeatherMapAPIHandler handler, HttpClient htt
         return HttpUtil.downloadImage(url);
     }
 
+    /**
+     * Get Weather data from the OneCall API for the given location. See https://openweathermap.org/api/one-call-api for
+     * details
+     * 
+     * @param location location represented as {@link PointType}
+     * @return
+     * @throws JsonSyntaxException
+     * @throws OpenWeatherMapCommunicationException
+     * @throws OpenWeatherMapConfigurationException
+     */
+    public synchronized @Nullable OpenWeatherMapOneCallAPIData getOneCallAPIData(@Nullable PointType location,
+            boolean excludeMinutely, boolean excludeHourly, boolean excludeDaily)
+            throws JsonSyntaxException, OpenWeatherMapCommunicationException, OpenWeatherMapConfigurationException {
+        Map<String, String> params = getRequestParams(handler.getOpenWeatherMapAPIConfig(), location);
+        StringBuilder exclude = new StringBuilder("");
+        if (excludeMinutely) {
+            exclude.append("minutely");
+        }
+        if (excludeHourly) {
+            exclude.append(exclude.length() > 0 ? "," : "").append("hourly");
+        }
+        if (excludeDaily) {
+            exclude.append(exclude.length() > 0 ? "," : "").append("daily");
+        }
+        logger.debug("Exclude: '{}'", exclude);
+        if (exclude.length() > 0) {
+            params.put(PARAM_EXCLUDE, exclude.toString());
+        }
+        return gson.fromJson(getResponseFromCache(buildURL(ONECALL_URL, params)), OpenWeatherMapOneCallAPIData.class);
+    }
+
+    /**
+     * Get the historical weather data from the OneCall API for the given location and the given number of days in the
+     * past.
+     * As of now, OpenWeatherMap supports this function for up to 5 days in the past. However, this may change in the
+     * future,
+     * so we don't enforce this limit here. See https://openweathermap.org/api/one-call-api for details
+     *
+     * @param location location represented as {@link PointType}
+     * @param days number of days in the past, relative to the current time.
+     * @return
+     * @throws JsonSyntaxException
+     * @throws OpenWeatherMapCommunicationException
+     * @throws OpenWeatherMapConfigurationException
+     */
+    public synchronized @Nullable OpenWeatherMapOneCallHistAPIData getOneCallHistAPIData(@Nullable PointType location,
+            int days)
+            throws JsonSyntaxException, OpenWeatherMapCommunicationException, OpenWeatherMapConfigurationException {
+        Map<String, String> params = getRequestParams(handler.getOpenWeatherMapAPIConfig(), location);
+        // the API requests the history as timestamp in Unix time format.
+        params.put(PARAM_HISTORY_DATE, Long.toString(new Date().getTime() / 1000 - days * 60 * 60 * 24));
```

Comment:
I guess this type of math can be done by Java and I am not sure if we should consider the timezone "UTC" here - something like:

```suggestion
        params.put(PARAM_HISTORY_DATE, Long.toString(ZonedDateTime.now().minusDays(days).toEpochSecond());
```

or

```suggestion
        params.put(PARAM_HISTORY_DATE, Long.toString(ZonedDateTime.now(ZoneId.of("UTC")).minusDays(days).toEpochSecond());
```

Reply:
I used the UTC version. Done